index.js

```diff
@@ -2,6 +2,11 @@ import { inactivityTime, urls } from "./src/shared.js";
 import { firebaseConfig as devFirebaseConfig } from "./src/dev/config.js";
 import { firebaseConfig as stageFirebaseConfig } from "./src/stage/config.js";
 import { firebaseConfig as prodFirebaseConfig } from "./src/prod/config.js";
+// When doing local development, uncomment this and comment out the line below it.
+// Get the API key file from Box or the DevOps team
+// Do not accept PRs with the localDevFirebaseConfig import uncommented
+// import { firebaseConfig as  localDevFirebaseConfig} from "./src/local-dev/config.js";
+const localDevFirebaseConfig = stageFirebaseConfig;
```

Comment:
I'm not sure I understand...does this point dev at the stage config until the developer comments it out and activates the line above?

What about just omitting this line `const localDevFirebaseConfig = stageFirebaseConfig;` and throwing a descriptive error in the else if block to avoid the config crossover?

```javascript
else if (isLocalDev) {
    if (!localDevFirebaseConfig) {
        throw new Error("localDevFirebaseConfig is missing. Update your config per team specs.");
    }
    !firebase.apps.length ? firebase.initializeApp(localDevFirebaseConfig()) : firebase.app();
}
```

Reply:
It was my experience that trying to require a nonexistent file resulted in an error where the page refused to load entirely, so it wouldn't allow it to get that far in the code.

Reply:
I see. That leads to a bit of a cryptic error either way. Here's the error when we set it to stage and try to log in.
<img width="442" alt="Screenshot 2024-09-04 at 10 45 30 AM" src="https://github.com/user-attachments/assets/3ad32fed-4dd1-448b-936e-12756309e44a">

My concern is that this error would pile on top of other hard-to-diagnose errors we've seen when auth goes down or other parts of the system get interrupted. I know it's a nitpick for the everyday case, but I think it helps diagnose the problem when we have more challenging access issues. I know there have been times when the firebase emulator integration for full stack work has taken hours to debug.

A type check on localDevFirebaseConfig does allow us to show a specific error message while omitting line 9 (the reference pointing from dev -> stage). My opinion is feel like that would be simpler and helpful for debugging when more complex access issues arise. 

```javascript
} else if (isLocalDev) {
        if (typeof localDevFirebaseConfig === 'undefined') {
            console.error('Local development requires a localDevFirebaseConfig function to be defined in src/local-dev/config.js.');
            return;
        }
        !firebase.apps.length ? firebase.initializeApp(localDevFirebaseConfig()) : firebase.app();
    } else {
```
<img width="466" alt="Screenshot 2024-09-04 at 11 07 00 AM" src="https://github.com/user-attachments/assets/e1b41d50-b3e1-45d3-bfd8-66a1a7097e52">

I'll approve and leave it up to you...that's my thought for all three integrations. Thanks